setup.py

```diff
@@ -277,6 +296,24 @@ def wrapper(*args, **kw):  # type: ignore
     _get_start_shard, lru_size=SLOTS_PER_EPOCH * 3)'''
 
 
+LIGHTCLIENT_PATCH_SUNDRY_FUNCTIONS = '''
+
+def get_generalized_index(ssz_class: Any, *path: Sequence[Union[int, SSZVariableName]]) -> GeneralizedIndex:
+    ssz_path = Path(ssz_class)
+    for item in path:
+        ssz_path = ssz_path / item
+    return GeneralizedIndex(ssz_path.gindex())
```

Comment:
Do we want generalized index logic to be "human readable"? If so, calling into a library that does weird class magic may not be the best choice and perhaps we could instead try to make it executable by pointing to https://github.com/ethereum/eth2.0-specs/blob/dev/ssz/merkle-proofs.md#ssz-object-to-index ?

Reply:
I opened #2179 to follow up.